src/calendar/view/eventeditor/AttendeeListEditor.ts

```diff
@@ -0,0 +1,277 @@
+								lang.get("passwordFor_label", {
+									"{1}": guest.address,
+								}),
+							helpLabel: () => m(".mt-s", m(CompletenessIndicator, { percentageCompleted: strength })),
+							key: address,
+							oninput: (newValue) => whoModel.setPresharedPassword(address, newValue),
+							injectionsRight: () => this.renderRevealIcon(guest.address),
+						})
+					})
+			: []
+
+		return m("", [...attendeeRenderers.map((r) => r()), externalGuestPasswords])
+	}
+
+	private renderRevealIcon(address: string): Children {
+		return m(IconButton, {
+			title: this.externalPasswordVisibility.get(address) === true ? "concealPassword_action" : "revealPassword_action",
+			click: () => {
+				this.externalPasswordVisibility.set(address, !this.externalPasswordVisibility.get(address))
+			},
+			icon: this.externalPasswordVisibility.get(address) === true ? Icons.NoEye : Icons.Eye,
+			size: ButtonSize.Compact,
+		})
+	}
+}
+
+/**
+ *
+ * @param editModel the event to set the organizer on when a button in the dropdown is clicked
+ * @param e
+ */
+function showOrganizerDropdown(editModel: CalendarEventWhoModel, e: MouseEvent) {
+	const lazyButtons = () =>
+		editModel.possibleOrganizers.map((organizer) => {
+			return {
+				label: () => organizer.address,
+				click: () => editModel.addAttendee(organizer.address, null),
+			}
+		})
+
+	createDropdown({ lazyButtons, width: 300 })(e, e.target as HTMLElement)
+}
+
+export function renderOrganizer(organizer: Guest, { model }: Pick<AttendeeListEditorAttrs, "model">): Children {
+	const { whoModel } = model.editModels
+	const { address, name, status } = organizer
+	const isMe = organizer.address === whoModel.ownGuest?.address
+	const editableOrganizer = whoModel.possibleOrganizers.length > 1 && isMe
+	const roleLabel = isMe ? `${lang.get("organizer_label")} | ${lang.get("you_label")}` : lang.get("organizer_label")
+	const statusLine = m(".small.flex.center-vertically", [renderStatusIcon(status), roleLabel])
+	const fullName = m("div.text-ellipsis", { style: { lineHeight: px(24) } }, name.length > 0 ? `${name} ${address}` : address)
+	const nameAndAddress = editableOrganizer
+		? m(".flex.flex-grow.items-center.click", { onclick: (e: MouseEvent) => showOrganizerDropdown(whoModel, e) }, [
+				fullName,
+				m(Icon, {
+					icon: BootIcons.Expand,
+					style: {
+						fill: theme.content_fg,
+					},
+				}),
+		  ])
+		: m(".flex.flex-grow.items-center", fullName)
+
+	const rightContent = isMe
+		? m(
+				"",
+				{ style: { minWidth: "120px" } },
+				m(DropDownSelector, {
+					label: "attending_label",
+					items: createAttendingItems(),
+					selectedValue: status,
+					class: "",
+					selectionChangedHandler: (value: CalendarAttendeeStatus) => {
+						if (value == null) return
+						whoModel.setOwnAttendance(value)
+					},
+				}),
+		  )
+		: m(IconButton, {
+				title: "sendMail_alt",
+				click: async () =>
+					(await import("../../../contacts/view/ContactView.js")).writeMail(
```

Comment:
oof this seems like something that should be somehow injected idk. maybe it's fine for now but is a bit dirty

Reply:
i felt it was too deeply nested to pass it all the way down

Reply:
just passing it from the editor dialog would be okay, I don't think deeply nested is a good counter argument but also it doesn't matter, if it's too much pain then just leave it